icann-rdap-client/src/md/redacted.rs

```diff
@@ -61,3 +68,501 @@ impl ToMd for &[Redacted] {
+                                        if x.is_string() {
+                                            match x.as_str() {
+                                                Some("") => Some(json!("*REDACTED*")),
+                                                Some(s) => Some(json!(format!("*{}*", s))),
+                                                _ => Some(json!("*REDACTED*")),
+                                            }
+                                        } else {
+                                            // it isn't a string, we aren't going to do anything with it
+                                            Some(final_value.clone()) // copy the found value back to it
+                                        }
+                                    },
+                                );
+                                // Now we check if we did something
+                                match replaced_json {
+                                    Ok(new_v) => {
+                                        *rdap_json_response = new_v; // we replaced something so now we need to update the response
+                                    }
+                                    _ => {
+                                        // Do nothing but we need to investigate why this is happening
+                                    }
+                                } // end match replace_with
+                            } // end if doing partialValue or emptyValue
+                        } // end if redaction_type
+                    } // end if final_path_option
+                } // end for each path_index_count
+            } // end if final_path
+        } // end loop over redactions
+    } // end if there is a redacted array
+}
+
+// This cleans it up into a json pointer which is what we need to use to get the value
+fn convert_to_json_pointer_path(path: &str) -> String {
+    let pointer_path = path
+        .trim_start_matches('$')
+        .replace('.', "/")
+        .replace("['", "/")
+        .replace("']", "")
+        .replace('[', "/")
+        .replace(']', "")
+        .replace("//", "/");
+    pointer_path
+}
+
+fn parse_redacted_array(
+    rdap_json_response: &Value,
+    redacted_array: &Vec<Value>,
+) -> Vec<RedactedObject> {
+    let mut list_of_redactions: Vec<RedactedObject> = Vec::new();
+
+    for item in redacted_array {
+        let item_map = item.as_object().unwrap();
+        let pre_path = item_map
+            .get("prePath")
+            .and_then(|v| v.as_str())
+            .map(|s| s.to_string());
+        let post_path = item_map
+            .get("postPath")
+            .and_then(|v| v.as_str())
+            .map(|s| s.to_string());
+        // this is the original_path given to us
+        let original_path = pre_path.clone().or(post_path.clone());
+        let mut redacted_object = RedactedObject {
+            name: Value::String(String::from("")), // Set to empty string initially
```

Comment:
is this not String::default()?

Reply:
Stylistic points? 😂  We can change that.